smart_open/s3.py

```diff
@@ -328,24 +339,24 @@ class SeekableBufferedInputBase(BufferedInputBase):
 
     Implements the io.BufferedIOBase interface of the standard library."""
 
-    def __init__(self, bucket, key, buffer_size=DEFAULT_BUFFER_SIZE,
+    def __init__(self, bucket, key, version_id=None, buffer_size=DEFAULT_BUFFER_SIZE,
                  line_terminator=BINARY_NEWLINE, session=None, resource_kwargs=None):
         if session is None:
             session = boto3.Session()
         if resource_kwargs is None:
             resource_kwargs = {}
         s3 = session.resource('s3', **resource_kwargs)
         self._object = s3.Object(bucket, key)
-
+        self._version_id = version_id
         try:
-            self._content_length = self._object.content_length
-        except botocore.client.ClientError:
-            raise ValueError(
-                '%r does not exist in the bucket %r, '
-                'or is forbidden for access' % (key, bucket)
-            )
-
-        self._raw_reader = SeekableRawReader(self._object, self._content_length)
+            self._content_length = _get(self._object, self._version_id)['ContentLength']
+        except botocore.client.ClientError as error:
+            raise IOError(str(error) + "\n" +
+                          '%r does not exist in the bucket %r, version_id %r'
```

Comment:
Please use hanging indent (see the previous version of the file for an example).